### PR TITLE
tests for #52003

### PR DESCRIPTION
**Pull request by ilrudie**
**Please provide a description of this PR:**

add unit test cases for #52003

### CODE REVIEW

File: pkg/config/analysis/analyzers/analyzers_test.go [hunk 1]
```diff
@@ -945,6 +945,22 @@ var testGrid = []testCase{
 			{msg.IneffectiveSelector, "Telemetry default/telemetry-ineffective"},
 		},
 	},
+	{
+		name:       "ServiceEntry Addresses Required Lower",
```
Comment (MorrisLaw) [inline pkg/config/analysis/analyzers/analyzers_test.go:949]: what do we mean by `lower` and `upper` in this context?

Comment (ilrudie) [reply pkg/config/analysis/analyzers/analyzers_test.go:949]: case used for `tcp` in protocol to test the fix

**Review by hzxuzhonghu — APPROVED** [review]
naming is a little confusing

**Comment by ilrudie** [comment]
> naming is a little confusing

Happy to adjust it in a follow up. Do you have any suggestions or a specific name which is off?

**Pull request merged**